backend/tests/courses/test_api.py

```diff
@@ -198,6 +202,77 @@ def test_search_by_instructor(self):
             )
 
 
+class CourseSearchRecommendationScoreTestCase(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        # Set up test data according to CourseRecommendationTestCase
+        CourseRecommendationsTestCase.setUpTestData()
+        cls.course_clusters = CourseRecommendationsTestCase.course_clusters
+
+    def setUp(self):
+        set_semester()
+        self.username = "jacob"
+        self.password = "top_secret"
+        self.user = User.objects.create_user(username=self.username, password=self.password)
+        self.client = APIClient()
+
+    def test_recommendation_is_null_when_course_not_part_of_model_even_when_logged_in(self):
+        self.client.login(username=self.username, password=self.password)
+
+        self.course, self.section = create_mock_data("PSCI-437-001", TEST_SEMESTER)
+        response = self.client.get(
+            reverse("courses-search", args=["current"]), {"search": "PSCI-437", "type": "auto"}
+        )
+
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(len(response.data), 1)
+        self.assertIs(response.data[0]["recommendation_score"], None)
+
+    def test_recommendation_is_null_when_user_not_logged_in(self):
+        response = self.client.get(
+            reverse("courses-search", args=["current"]), {"search": "PSCI", "type": "auto"}
+        )
+
+        self.assertEqual(200, response.status_code)
+        for course in response.data:
+            self.assertEqual(course["recommendation_score"], None)
+
+    @patch(
+        "courses.views.CourseListSearch.get_serializer_context",
+        new=production_CourseListSearch_get_serializer_context,
+    )
+    @patch("courses.views.retrieve_course_clusters")
+    def test_recommendation_is_number_when_user_is_logged_in(self, course_clusters_mock):
+        course_clusters_mock.return_value = self.course_clusters
+        self.client.login(username=self.username, password=self.password)
+
+        curr_semester_schedule = Schedule.objects.create(
+            person=self.user, name="curr_semester_schedule", semester=TEST_SEMESTER
+        )
+        curr_semester_schedule.save()
+        # NOTE: many of the sections in this schedule match up with the `semester`
+        # used to create the schedule
+        curr_semester_schedule.sections.add(
+            Course.objects.get(full_code="PSCI-498", semester="2019A").sections.get(code="001")
```

Comment:
Shouldn't the semester of each course in the schedule match the semester of the schedule? Not sure I understand the comment/note above.

Reply:
That comment should read
> NOTE: many of the sections in this schedule **do not** match up with the `semester` **of the schedule**

I added it because we generally try to maintain the invariant that the schedule semester matches the semester of all the courses in the schedule. However `TEST_SEMESTER`, which is the variable that we used for the schedule's semester, does not necessarily match up with the course semesters that we place in the schedule.

Reply:
Ok sounds good